### PR TITLE
New tutorial: Managing High-Volume Batch Processing Using Amazon SQS and EKS

### PR DESCRIPTION
**Pull request by tucktuck9**
# Content Pull Request Checklist

Please check to make sure your content submission meets the following requirements by adding an `x` between the brackets:

- [X] The title of this pull request reflects the title of my content.
- [X] I have reviewed the [Content Review Checklist](/content/blob/main/CONTENT_REVIEW_CHECKLIST.md).
- [X] This pull request contains one piece of content (post or page).

**By submitting this pull request, I confirm I own this content, have permission to use each image included in this content, and that it will be released under the [CC BY-SA 4.0 SA License](/LICENSE).**

### CODE REVIEW

**Review by DalM2023 — COMMENTED** [review]
1st round edit finished. Ready for review.

File: tutorials/managing-high-volume-batch-sqs-eks/index.md [hunk 1]
```diff
@@ -0,0 +1,604 @@
...
+
+In this tutorial, you'll integrate Amazon SQS with your Amazon EKS cluster, build a batch processing application, containerize the application and deploy to Amazon ECR, then use an Amazon SQS job queue to run your batch tasks. In the second half of the tutorial, we'll shift gears to the EFS CSI Driver, which allows us to keep our data intact across multiple nodes while running batch workloads.
```
[the remark this answers is not in the document]
Comment (tucktuck9) [reply tutorials/managing-high-volume-batch-sqs-eks/index.md]: Updated, and deleted 'In the second half of the tutorial, we'll shift gears to the EFS CSI Driver, which allows us to keep our data intact across multiple nodes while running batch workloads.'

File: tutorials/managing-high-volume-batch-sqs-eks/index.md [hunk 2]
```diff
@@ -0,0 +1,604 @@
...
+
+With the completion of this tutorial, you've successfully orchestrated batch processing tasks in your Amazon EKS cluster using Amazon SQS and EFS. You've not only integrated SQS as a robust job queue but also leveraged the EFS CSI Driver for persistent storage across multiple nodes. This tutorial has walked you through the setup of your Amazon EKS cluster, the deployment of a Python-based batch processing application, and its containerization and storage in Amazon ECR. You've also learned how to create multi-architecture images and deploy them as Kubernetes Jobs. Furthermore, you've extended the capabilities of your Kubernetes Jobs by integrating them with Amazon SQS and providing persistent storage through Amazon EFS. To continue your journey, setup the [Cluster Autoscaler](https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler) for dynamic scaling, or explore [EFS Lifecycle Management](https://docs.aws.amazon.com/efs/latest/ug/lifecycle-management-efs.html) to automate moving files between performance classes or enabling [EFS Intelligent-Tiering](https://docs.aws.amazon.com/efs/latest/ug/intelligent-tiering.html) to optimize costs.
+
+## Clean up
```
[the remark this answers is not in the document]
Comment (tucktuck9) [reply tutorials/managing-high-volume-batch-sqs-eks/index.md]: Updated.

File: tutorials/managing-high-volume-batch-sqs-eks/index.md [hunk 3]
```diff
@@ -0,0 +1,604 @@
...
+```
+
+## Conclusion
+
+With the completion of this tutorial, you've successfully orchestrated batch processing tasks in your Amazon EKS cluster using Amazon SQS and EFS. You've not only integrated SQS as a robust job queue but also leveraged the EFS CSI Driver for persistent storage across multiple nodes. This tutorial has walked you through the setup of your Amazon EKS cluster, the deployment of a Python-based batch processing application, and its containerization and storage in Amazon ECR. You've also learned how to create multi-architecture images and deploy them as Kubernetes Jobs. Furthermore, you've extended the capabilities of your Kubernetes Jobs by integrating them with Amazon SQS and providing persistent storage through Amazon EFS. To continue your journey, setup the [Cluster Autoscaler](https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler) for dynamic scaling, or explore [EFS Lifecycle Management](https://docs.aws.amazon.com/efs/latest/ug/lifecycle-management-efs.html) to automate moving files between performance classes or enabling [EFS Intelligent-Tiering](https://docs.aws.amazon.com/efs/latest/ug/intelligent-tiering.html) to optimize costs.
```
[the remark this answers is not in the document]
Comment (tucktuck9) [reply tutorials/managing-high-volume-batch-sqs-eks/index.md]: Updated.

File: tutorials/managing-high-volume-batch-sqs-eks/index.md [hunk 4]
```diff
@@ -0,0 +1,604 @@
+---
+title: "Managing High-Volume Batch Processing Using Amazon SQS and EKS"
+description: "How to run robust, scalable, and data-heavy batch processing tasks in your Amazon EKS cluster using Amazon SQS for decoupled job orchestration and Amazon EFS for persistent, scalable storage."
+tags:
```
[the remark this answers is not in the document]
Comment (tucktuck9) [reply tutorials/managing-high-volume-batch-sqs-eks/index.md:4]: Updated.

File: tutorials/managing-high-volume-batch-sqs-eks/index.md [hunk 5]
```diff
@@ -0,0 +1,604 @@
...
+ --name ecr-sa \
+ --namespace default \
+ --cluster batch-quickstart \
+ --attach-policy-arn arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly \
+ --approve
+```
+
+The EFS CSI Driver does not have an AWS managed policy, so there are a few additional steps to create the service account. For instructions, see [Designing Scalable and Versatile Storage Solutions on Amazon EKS with the Amazon EFS CSI](https://quip-amazon.com/KQAIAANyMa13).
+
+## Step 3: Verify the EFS CSI Driver Add-On is Installed
```
[the remark this answers is not in the document]
Comment (tucktuck9) [reply tutorials/managing-high-volume-batch-sqs-eks/index.md]: Updated.

File: tutorials/managing-high-volume-batch-sqs-eks/index.md [hunk 6]
```diff
@@ -0,0 +1,604 @@
+---
```
[the remark this answers is not in the document]
Comment (tucktuck9) [reply tutorials/managing-high-volume-batch-sqs-eks/index.md:1]: Thank you @DalM2023! I incorporated all of your changes, as well as made some significant revisions to the tutorial title and introduction. Largely based on both your feedback and late-to-the-party reviewers. Would you please take another look.

**Comment by scribwell** [comment]
@DalM2023 The author has finished their revisions and this tutorial is ready for your second round of review.

File: tutorials/managing-high-volume-batch-sqs-eks/index.md [hunk 7]
```diff
@@ -0,0 +1,604 @@
...
... alignment, and variant calling, but the actual genomic data files far exceed SQS's size constraints. This is where the integration with Amazon EFS becomes invaluable. By using SQS messages to contain metadata or pointers to large data files stored in a scalable and secure EFS filesystem, you achieve several key benefits. You decouple data processing steps, allowing each component of your pipeline to operate independently. You can also scale horizontally as EFS's architecture grows with your data, eliminating the need for downtime or complex resizing operations. EFS provides a persistent storage layer for large workloads, ensuring that processed data from one job can be easily accessed by subsequent jobs or different applications. The integration unifies your operations by letting SQS manage job orchestration and EFS handle data storage, streamlining system monitoring, logging, and troubleshooting. In essence, combining SQS and EFS provides a robust, scalable, and well-rounded solution for handling data-heavy batch processing tasks in Amazon EKS, ideal for a wide range of applications from complex genomic analyses to large-scale machine learning projects and sophisticated ETL jobs.
```
[the remark this answers is not in the document]
Comment (tucktuck9) [reply tutorials/managing-high-volume-batch-sqs-eks/index.md]: Made significant changes to tutorial introduction, not done/requested re-review

File: tutorials/managing-high-volume-batch-sqs-eks/index.md [hunk 8]
```diff
@@ -0,0 +1,604 @@
+---
+title: "Managing High-Volume Batch Processing Using Amazon SQS and EKS"
+description: "How to run robust, scalable, and data-heavy batch processing tasks in your Amazon EKS cluster using Amazon SQS for decoupled job orchestration and Amazon EFS for persistent, scalable storage."
```
[the remark this answers is not in the document]
Comment (tucktuck9) [reply tutorials/managing-high-volume-batch-sqs-eks/index.md]: Made significant changes to tutorial title and subtitle, not done/requested re-review

File: tutorials/managing-high-volume-batch-sqs-eks/index.md [hunk 9]
```diff
@@ -0,0 +1,604 @@
+---
+title: "Managing Asynchronous Tasks with SQS and EFS Persistent Storage in Amazon EKS"
+description: "How to run asynchronous background tasks in your Amazon EKS cluster using Amazon SQS for task orchestration and Amazon EFS for scalable, high-availability storage."
```
Comment (DalM2023) [inline tutorials/managing-high-volume-batch-sqs-eks/index.md:3]: The description can host some 60 characters or fewer. It shouldn't repeat what's in the title. Instead, it should highlight the reason why someone would want to read the blog such as the problem it's solving or what the reader will learn. 

Consider rewriting. Here's an example: Run background tasks for orchestration and high availability storage.

File: tutorials/managing-high-volume-batch-sqs-eks/index.md [hunk 10]
```diff
@@ -0,0 +1,604 @@
...
+  --region ${CLUSTER_REGION} \
+  --cluster batch-quickstart \
+  --namespace default \
+  --name ecr-sa \
+  --attach-policy-arn arn:aws:iam::aws:policy/AmazonSQSFullAccess \
+  --override-existing-serviceaccounts \
+  --approve
+```
+
+## Step 9: Create a Kubernetes secret
```
Comment (DalM2023) [inline tutorials/managing-high-volume-batch-sqs-eks/index.md:368]:
```suggestion
## Step 9: Create a Kubernetes Secret
```

File: tutorials/managing-high-volume-batch-sqs-eks/index.md [hunk 11]
```diff
@@ -0,0 +1,604 @@
...
+-n default
+```
+
+The expected output should look like this:
+
+```
+secret/regcred created
+```
+
+## Step 10: Deploy the Kubernetes Job with Queue Integration
```
Comment (DalM2023) [inline tutorials/managing-high-volume-batch-sqs-eks/index.md:394]:
```suggestion
## Step 10: Deploy the Kubernetes Job With Queue Integration
```

File: tutorials/managing-high-volume-batch-sqs-eks/index.md [hunk 12]
```diff
@@ -0,0 +1,604 @@
...
+```
+
+The expected output should look like this:
+
+```
+persistentvolume/efs-pv created
+persistentvolumeclaim/efs-claim created
+```
+
+## Step 12: Implement Persistent Storage with Amazon EFS
```
Comment (DalM2023) [inline tutorials/managing-high-volume-batch-sqs-eks/index.md:503]:
```suggestion
## Step 12: Implement Persistent Storage With Amazon EFS
```

File: tutorials/managing-high-volume-batch-sqs-eks/index.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (DalM2023) [reply tutorials/managing-high-volume-batch-sqs-eks/index.md:1]: @tucktuck9 The second version reads much, much better! Good job with the rewrite.  

I picked up on a few minor edits. Please see below. But overall, this piece is in better shape.

**Review by DalM2023 — COMMENTED** [review]
2nd edit completed. Ready for review.

File: tutorials/managing-high-volume-batch-sqs-eks/index.md (hunk 10 above)
Comment (tucktuck9) [reply tutorials/managing-high-volume-batch-sqs-eks/index.md]: Updated.

File: tutorials/managing-high-volume-batch-sqs-eks/index.md (hunk 11 above)
Comment (tucktuck9) [reply tutorials/managing-high-volume-batch-sqs-eks/index.md]: Updated.

File: tutorials/managing-high-volume-batch-sqs-eks/index.md (hunk 12 above)
Comment (tucktuck9) [reply tutorials/managing-high-volume-batch-sqs-eks/index.md]: Updated.

File: tutorials/managing-high-volume-batch-sqs-eks/index.md (hunk 9 above)
Comment (tucktuck9) [reply tutorials/managing-high-volume-batch-sqs-eks/index.md]: Updated.

File: tutorials/managing-high-volume-batch-sqs-eks/index.md [hunk 13]
```diff
@@ -0,0 +1,604 @@
...
+```
+# Delete the SQS Queue
+aws sqs delete-queue --queue-url YOUR_SQS_QUEUE_URL
+
+# Delete the ECR Repository
+aws ecr delete-repository --repository-name YOUR_ECR_REPO_NAME --force
+```
+
+If you enjoyed this tutorial, found any issues, or have feedback for us, <a href="https://www.pulse.aws/survey/Z8XBGQEL" target="_blank">please send it our way!</a>
+## Conclusion
```
Comment (DalM2023) [inline tutorials/managing-high-volume-batch-sqs-eks/index.md:600]: @tucktuck9 Should there be a space between 599 and 600?

File: tutorials/managing-high-volume-batch-sqs-eks/index.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (DalM2023) [reply tutorials/managing-high-volume-batch-sqs-eks/index.md:1]: @tucktuck9 Almost there! Well done on descriptor rewrite. I saw something between lines 599 and 600... it's a formatting issue. I flagged it below. Other than that, it's in good shape.

**Review by DalM2023 — COMMENTED** [review]
3rd edit completed. Ready for review.

File: tutorials/managing-high-volume-batch-sqs-eks/index.md (hunk 13 above)
Comment (tucktuck9) [reply tutorials/managing-high-volume-batch-sqs-eks/index.md:601]: Updated.

**Comment by tucktuck9** [comment]
> 3rd edit completed. Ready for review.

Updated - Added a space between survey link and conclusion.

**Review by damonk10 — APPROVED** [review]

**Pull request merged**